### ISSUE TITLE
Decide whether to support Android 8.0+

### ISSUE THREAD

**Issue by smichel17** [LibreShift/red-moon#169]
As of Android 8.0, there are major changes to overlays [(source)](https://developer.android.com/about/versions/oreo/android-8.0-changes.html). Notably, we won't be able to display an overlay over keyboards (or other input devices). Given that Android 8.0 already has a blue light filter, we need to decide whether it is worth the hassle of supporting overlay mode on these devices.

**Comment by mueller-ma** [comment]
You should wait until some companies made their Android 8 rom and see if they removed the blue filter

**Comment by smichel17** [comment]
Yeah, this decision is a ways off anyway. I'm going to wait until Red Moon is feature- complete to worry about it.

**Comment by kalvn** [comment]
Just got Oreo on my Nexus 5X and the blue light filter doesn't seem to be available. Looks like it's only for Pixel phones for now. So I would definitely appreciate Oreo support :)

**Comment by smichel17** [comment]
@kalvn Out of curiosity, does red moon work at all on Oreo right now?

**Comment by kalvn** [comment]
@smichel17 no, it crashes immediately.

**Comment by alexthomazo** [comment]
Hello,

I had the update 3 days ago on my Nexus 5X as well, working with version 3.2.0 except for the notification area (closed or opened) and for the lock screen.

Everything else is working as excepted, even the keyboard.

As they switched the notification area with white background, the light is very bright when red moon is activated :(

**Comment by kalvn** [comment]
@alexthomazo I agree but I think nothing can be done for now. I miss the previous dark notification bar :/

I noticed a problem though: the schedule doesn't seem to work anymore. Am I the only one with this issue?

**Comment by smichel17** [comment]
> The schedule doesn't seem to work anymore

Is this on 8.0? If so, you're out of luck for the moment, sorry. As noted in this issue, 8.0 breaks a ton of things and will be a lot of effort to add support for; I'm definitely not going to try until Red Moon is feature-complete (in a few months, hopefully), and I'll re-evaluate at that point.

If on an earlier android version, I'm aware that it *was* broken, but though I'd finally fixed it in v3.2.0 (see #144).

**Comment by smichel17** [comment]
@kalvn try turning the schedule off and on again, see #194

**Comment by kalvn** [comment]
@smichel17 it worked, thanks :)

**Comment by Madis0** [comment]
The keyboard is tinted and so are the secure views (permissions, etc), only thing that is not is the notification bar and shade. 

Pretty much only things you could do to improve the implementation are 
- adding notification channels, so that their properties can be managed in OS 
- if possible somehow, make all apps use a dark-on-white notification bar, to make it less in user's face

**Comment by thomasaiman** [comment]
How are you feeling about Oreo these days? I finally got it rolled out to my device (Motorola G5 Plus) and there's no "night light" available from the OS. Red Moon is still missing the shade and notification bar, but otherwise works fine.

**Comment by smichel17** [comment]
My AT&T Galaxy S6 (which I quite dislike, for the record; it's far too locked down) is still on Nougat (7.0). It has the night light feature, which I use. I also use Red Moon on top of that, but only the dimming functionality. If root mode were done and I had root access on this phone (grr), I'd probably use Red Moon instead of night light so I could make the effect even stronger ;).

As for supporting Oreo, it's in the same state as the rest of the project: I'd like to do it some day, but there's a lot of [other things](https://snowdrift.coop) I want to work on, too, so Red Moon is mostly in bugfix mode until further notice (although I do plan to do a release soon, mostly to add updated translations, and I would be up for putting more time in if someone wanted to join me as a co-maintainer -- see my [note in the README](https://github.com/LibreShift/red-moon#looking-for-co-maintainers)).

**Comment by smichel17** [comment]
Thanks to @AdamNiederer in #221, Oreo is *supported*; that's just been released and will show up in F-Droid soon.

There's a separate question of targeting Oreo (bumping the `targetSdkVersion` from 25 to 26). Most of my reservations about supporting Oreo were due to extra work that will be required in order to do that. However, Google has made that choice for us: the Play Store will soon (maybe already?) require apps to target api 26.

I'm not planning on doing this right away, mostly because F-Droid still works, and there are other issues with the Play Store listing right now (#148). I opened #242 to track it.
